WordPress/tumblr-importer
# use composer/autoload

### what/why

**The purpose of this PR is to add support for composer/autoload.**

- added entries for prod usage in `composer.json`, also added [redvelvet-lib](https://github.a8c.com/Automattic/redvelvet-lib) as dependecy (not in use), also added some handy scripts.
- added script to generate a `.zip` file
- added a new workflow file to generate a release `build-release.yml`
- changed classes and `tumblr-importer.php` to use the composer auto-loader
- changed folder structure, follow the new structure:
```
.
├── composer.json
├── composer.lock
├── languages
│   └── tumblr-importer.pot
├── license.txt
├── readme.txt
├── scripts
│   └── build-release.sh
├── src
│   ├── class-tumblr-import.php
│   ├── class-wp-importer-cron.php
│   └── index.php
├── tumblr-importer.php
└── vendor
    ├── ...
```

### testing

I've written some comments throughout the files to help the reviewers and also I couldn't debug the cron execution to ensure everything is ok.

- git checkout this branch
- run `sh scripts/build-release.sh`
- install the plugin using the generated `tumblr-importer.zip` 
- follow the regular import flow steps
